src/common/snippets/include/snippets/op/subgraph.hpp

```diff
@@ -116,6 +120,7 @@ class Subgraph : public ngraph::op::Op {
     void print_statistics(bool verbose);
 
     void serialize() const;
+    void set_master_shape(ov::PartialShape new_shape) {master_shape = std::move(new_shape);}
```

Comment:
public `master_shape` setter is smt not obvious. My understanding was: master shape is computed inside subgraph == full under Subgraph control.

Reply:
Completely agree, but this what we have to pay for the plugin-side dimension collapsing. I believe we can remove it in the transpose PR, where we start using rt_info for shapes communication. Let's discuss it then.